### DISCUSSION TITLE
Conveyor Belt Tomography

### DISCUSSION THREAD

**Discussion by aenne-abel** [astra-toolbox/astra-toolbox#392] [Q&A]
Hi all,
I'm currently trying to simulate a conveyor belt tomography setup.
I effectively have a 4 pixel detector, simulated with GEANT4, and a point source emmitting 4.44 MeV gammas. The detector and source are positioned along the z-axis. The object then moves along the x-axis in between the source and detector in 360 steps. After each step, the object is turned by 1°. My object is a simple 5x5x5 cm aluminium cube. 0° corresponds to the edges of the object being parallel to the x-, y- and z-axis

I tried the same simulation with fixed object positions before and the tomography turned out fine. My sinogram also looks reasonable:

![Sinogram_ConveyorBelt](https://user-images.githubusercontent.com/107397117/199501207-97bf3c12-acd4-4d11-a669-a0e970974ff3.jpeg)

I streched the x-axis a little in order to better see the sinogram. As you can see for the first few projections the object is not blocking any particles and thus no absorption is visible. At roughly projection number 150 and 210 a maximal absorption is visible because the object absorbs a high fraction of particles because on one hand it is positioned right between the detector and source and additionally, due to the turning, the particles travel through a lot of material.

However I can't seem to be able to produce a sensible tomography image. 

![Tomography_ConveyorBelt](https://user-images.githubusercontent.com/107397117/199501347-f9d1c31c-e6a7-4c2f-a08c-01d5d8a74c50.jpeg)

This is what it looks like right now reconstructed with SIRT_CUDA and 100 iterations. I would assume that my positioning in ASTRA Toolbox should be correct as I previously also worked with an object that was shifted and turned. My projection geometry is calculated as follows:

	detPixelSizeAtObj_cm = binwidth_det*float(source_dist)/(float(source_dist)+float(det_dist))	# size of one detector pixel at object position
	eff_det_size = float(xMax)-float(xMin) + nDetColumns*detPixelSizeAtObj_cm	# size of the whole detector at the object position
	size_pixel = eff_det_size/nVoxel	# size of one image pixel (voxel) in cm
	binwidthDetInPixel = binwidth_det/size_pixel	# binwidth of the detector in image pixels
	positions = positions/size_pixel	# in object positions pixel

	vol_geom = astra.create_vol_geom(nVoxel, nVoxel)

	vec_S = np.array([0,-float(source_dist)])/size_pixel # positions of source in pixel
	vec_D = np.array([0,float(det_dist)+1])/size_pixel # position of detector in pixel
	vec_U = np.array([binwidthDetInPixel,0]) # detector vector along detector pixels in image pixel
	vectors = np.zeros((nSteps,6))
	for i in range(0,nSteps):
		curr_ang = angles[i]	# turning angle of object
		M_R = np.array([[np.cos(curr_ang), -np.sin(curr_ang)],[np.sin(curr_ang), np.cos(curr_ang)]])	# rotation matrix
		the_shift = positions[i]	# object shift in pixel

		vectors[i,0:2] = np.dot(M_R,vec_S)+the_shift*np.dot(M_R,vec_U)/binwidthDetInPixel	# position of source in coordinate system of the object in image pixels
		vectors[i,2:4] = np.dot(M_R,vec_D)+the_shift*np.dot(M_R,vec_U)/binwidthDetInPixel	# position of detector in corrdinate system of the object
		vectors[i,4:] = np.dot(M_R,vec_U)	# detector vector in coordinate system of the object

	proj_geom = astra.create_proj_geom('fanflat_vec', nDetColumns, vectors)

Can anybody help me with fixing my tomography? Do I need significantly more projections when using a conveyor belt setup? Is there a specific reconstruction algorithm? (A couple different ones I've tried didn't give a much better result.) Is a conveyor belt setup even possible with ASTRA Toolbox?
I would greaty appreciate any advice/experience with this kind of tomography setup.

**Comment by askorikov** [comment]
Hi! If this is still relevant, can you provide the values for your geometry parameters (`binwidth_det`, `source_dist`, etc.) so that we can reproduce your example exactly?

**Comment by aenne-abel** [comment]
Hi, yes I'm still interested! The distance of the source to the detector is 45 cm, the distance from the source to the center of the object is 37.6 cm. The object is a 5x5x5 cm cube. The bin width of the detector pixel is 6.2 mm. In total 16 detector pixel arranged in a 4x4 grid were simulated, but for the reconstruction the counts from each y-row were combined  (reconstruction in the x-z-plane, particles travelling in z-direction). The object was moved between the source and the detector from x = -3.0407 cm to x = 3.0407 cm, perpendicular to the beam direction. In total 360 projections were taken with an equal step size in x-direction. After every projection, the object was moved and then turned by 1° around the y-axis. The object is not a perfect absorber and the detector is not ideal. Therefore scattering processes are included in the absorption values shown in the sinogram.
If you need any more information, let me know.

**Comment by askorikov** [comment]
I haven't had a chance to look at it in detail yet, but I wonder: have you already seen the conveyor belt tomography section in the original Astra paper (https://repository.uantwerpen.be/docman/irua/6cb7dd/139216.pdf)?

**Comment by aenne-abel** [comment]
Hi sorry for the late reply. I did take a look at the Astra paper, this was in fact the motivation to try a conveyor belt setup in the first place.

**Comment by askorikov** [comment]
I've had a chance to look at it again, and geometry definition seems fine to me. May there is some discrepancy between the geometry definition that you use in Astra and in GEANT4? When I use projections simulated in Astra using the same geometry, this is the reconstruction I get:
![image](https://github.com/astra-toolbox/astra-toolbox/assets/31259212/ec89ad86-5e34-4ef3-b78f-478bf51eccbb)
I contains a lot of artifacts from the conveyor belt geometry, but still the correspondence seems to be much better than in your experiment.
